### DISCUSSION TITLE
How to use appargs

### DISCUSSION THREAD

**Discussion by FrenchGithubUser** [wailsapp/wails#2415] [Q&A]
I have tried to use the appargs variable in the wails.json config file like this : 

"appargs": "mode=dev"

I also tried starting wails with args like this : wails dev -appargs "mode=dev".

Then, the args are read in the main.go file (and stored in a variable) : 

flag.Parse()
variables.ARGS = flag.Args()

But the variable is always empty. What am I doing wrong ?

**Comment by ianmjones** [comment]
Can you show a little more code, including when you read `os.Args` and pass them into `flag.Parse()`?

**Comment by FrenchGithubUser** [comment]
The issue came from the fact that I was trying to read the flags in the main function. For some reason, the code was not executed there. So I now read them in the NewApp() function. Is it considered as the way to do it ?

**Comment by adlion** [comment]
I am confused with args

Can someone help please

I parse flags ar main.go using 

```go
lclConfigFlag := flag.String("config", "", "Configuration file.")
	flag.Parse()
	fmt.Println("Config flag value:", *lclConfigFlag)
```

and execute 
```bash
wails dev  -appargs "-config SOME_FILE"
wails dev --appargs "-config SOME_FILE"
wails dev -appargs "config SOME_FILE"
wails dev -appargs "--config SOME_FILE"
wails dev -appargs "config SOME_FILE"
```

If I build I can pass the flag successfully
